src/cdk/overlay/overlay-directives.ts

```diff
@@ -78,22 +77,21 @@ export const MAT_CONNECTED_OVERLAY_SCROLL_STRATEGY_PROVIDER = {
   selector: '[cdk-overlay-origin], [overlay-origin], [cdkOverlayOrigin]',
   exportAs: 'cdkOverlayOrigin',
 })
-export class OverlayOrigin {
+export class CdkOverlayOrigin {
```

Comment:
I was thinking about this one a few days ago: do we really need the directive? It doesn't really provide a lot of value since using it ends up being slightly longer than straight-up passing in the `elementRef` to the overlay:

```html
<button cdk-overlay-origin #origin="cdkOverlayOrigin">Open</button>
<ng-template cdk-connected-overlay [origin]="origin"></ng-template>
```

Versus something like this if we got rid of it:

```html
<button #origin>Open</button>
<ng-template cdk-connected-overlay [origin]="origin"></ng-template>
```

It's not something we should necessarily address now, but it just something to think about.

Reply:
The problem is that if the element you're applying it do has a component (like `mat-button`), you don't have a way to grab the ElementRef in the template.